src/c++/perf_analyzer/genai-pa/genai_pa/llm_profile.py

```diff
@@ -0,0 +1,146 @@
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import json
+from dataclasses import dataclass
+from itertools import pairwise
+
+import numpy as np
+from transformers import AutoTokenizer
+
+
+@dataclass
+class LLMMetrics:
+    time_to_first_tokens: list[int]
+    inter_token_latencies: list[int]
+    output_token_throughputs: list[int]
+
+    def get_base_name(self, attr_name: str) -> str:
+        if attr_name == "time_to_first_tokens":
+            return "time_to_first_token"
+        elif attr_name == "inter_token_latencies":
+            return "inter_token_latency"
+        elif attr_name == "output_token_throughputs":
+            return "output_token_throughput"
+        else:
+            raise ValueError(f"No attribute named '{attr_name}' exists.")
+
+
+class Statistics:
+    # TODO: make this parameter LLM agnostic
+    def __init__(self, metrics: LLMMetrics):
+        # iterate through LLMMetrics to calculate statistics and set attributes
+        for attr, data in metrics.__dict__.items():
+            attr = metrics.get_base_name(attr)
+            self._calculate_mean(data, attr)
+            self._calculate_percentiles(data, attr)
+            self._calculate_minmax(data, attr)
+            self._calculate_std(data, attr)
+
+    def _calculate_mean(self, data: list[int], attr: str):
+        avg = np.mean(data)
+        setattr(self, "avg_" + attr, avg)
+
+    def _calculate_percentiles(self, data: list[int], attr: str):
+        p50, p90, p95, p99 = np.percentile(data, [50, 90, 95, 99])
+        setattr(self, "p50_" + attr, p50)
+        setattr(self, "p90_" + attr, p90)
+        setattr(self, "p95_" + attr, p95)
+        setattr(self, "p99_" + attr, p99)
+
+    def _calculate_minmax(self, data: list[int], attr: str):
+        min, max = np.min(data), np.max(data)
+        setattr(self, "min_" + attr, min)
+        setattr(self, "max_" + attr, max)
+
+    def _calculate_std(self, data: list[int], attr: str):
+        std = np.std(data)
+        setattr(self, "std_" + attr, std)
+
+    def __repr__(self):
+        attr_str = ""
+        for k, v in self.__dict__.items():
+            attr_str += f"{k}={v},"
+        return f"Statistics({attr_str})"
+
+
+class LLMProfileData:
+    def __init__(self, filename: str, tokenizer_model: str = "gpt2") -> None:
```

Comment:
can we pass in the tokenizer itself instead of the model?

Reply:
Any reason for the higher level script to provide the tokenizer instance? It would make sense if the higher script has use cases for tokenizer. The main reason for instantiating tokenizer inside `LLMProfileData` was to use it for calculating token throughput which only happens inside the class.

Reply:
We eventually want to tokenizer the requests too. So eventually we will need it as a separate module/method.

Reply:
@tgerdesnv if you are good with it, lets punt on this ai. I wanna get this in and @nv-hwoo will take on the tokenizer module next.

Reply:
The reason to be passed in is because we would want to use the same tokenizer for inputs and outputs, and also because the user may supply their own

Reply:
I will set it to take tokenizer for now and we can change later depending on later use case. If the higher level script uses tokenizer as well, I think it makes sense to pass down one to the class.

Reply:
Cool cool.